### PR TITLE
feat(ImageReslice): Add vtkImageReslice filter

### PR DESCRIPTION
**Pull request by finetjul**
This is a limited implementation of vtkImageReslice.
It does not support:
 - LINEAR or CUBIC interpolation.
 - Slab mode
 - Permute optimization
 - Convenient reslice axes methods

### CODE REVIEW

**Comment by finetjul** [comment]
@agirault can you please review ?

**Comment by laurennlam** [comment]
@jourdain Would it be possible to merge it as there are only new files which don't affect the actual vtk.js version ? Because we really need it for a customer.

File: Sources/Imaging/Core/AbstractImageInterpolator/index.js [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+const DEFAULT_VALUES = {
+  outValue: 0,
+  tolerance: Number.EPSILON,
+  componentOffset: 0,
+  componentCount: -1,
+  borderMode: ImageBorderMode.CLAMP,
+  slidingWindow: false,
+
+  scalars: null,
+  interpolationInfo: Object.assign({}, vtkInterpolationInfo),
```
Comment (jourdain) [inline Sources/Imaging/Core/AbstractImageInterpolator/index.js]: I'm guessing that you know that all the arrays inside are just reference to a single instance of those array?

Comment (finetjul) [reply Sources/Imaging/Core/AbstractImageInterpolator/index.js]: Yes, I use `subarray` as an address pointer. I don't want to copy the array only to point at different places in the array.

Comment (jourdain) [reply Sources/Imaging/Core/AbstractImageInterpolator/index.js]: subarray will recreate a new array...

Comment (finetjul) [reply Sources/Imaging/Core/AbstractImageInterpolator/index.js]: It's a new array but it points on the same array buffer. So it is fast (no copy)

Comment (jourdain) [reply Sources/Imaging/Core/AbstractImageInterpolator/index.js]: Ok I was talking about the array you define by [] inside vtkInterpolationInfo.

**Pull request merged**